libs/api/src/lib/graphql/phrase/phrase.public-queries.ts

```diff
@@ -0,0 +1,103 @@
+import {PrismaClient} from '@prisma/client'
+import {articleWithRevisionsToPublicArticle} from '../../db/article'
+import {pageWithRevisionsToPublicPage} from '../../db/page'
+import {Context} from '../../context'
+
+export const queryPhrase = async (
+  query: string,
+  prisma: PrismaClient,
+  publicArticlesLoader: Context['loaders']['publicArticles'],
+  publicPagesLoader: Context['loaders']['publicPagesByID']
+) => {
+  const [articleRevisions, pageRevisions] = await Promise.all([
+    prisma.$queryRaw<{id: string}[]>`
```

Comment:
This is very expensive, if a lot of revisions exist. Couldn't we just search in the latest revision?

Reply:
I tested it with 100k revisions and was fine due to the index. The problem with only searching for published revisions is that I need to join another table which was slower in my testing :(

Reply:
Very interesting!